app.py

```diff
@@ -80,19 +79,65 @@ def get_all_routes():
 
 @app.route("/id", methods=['GET'])
 def get_id_for_route():
-	id = session.query(db.Route).filter(db.Route.name == name).one().id
+	id = session.query(db.Route).filter(db.Route.name == request.args['name']).one().id
 	return json.dumps(id)
 
-
 @app.route("/station/<string:station_id>", methods=['GET'])
 def get_station_details(station_id):
 	station = session.query(db.Station).filter(db.Station.id == station_id).one()
-
 	route_name = session.query(db.Route).filter(db.Route.id == station.route_id).one().name
+	
+	# add in predictions for both directions
+	stations_details = {'name': station.name_human_readable, 
+						'route_name': route_name, 
+						'id': station.id}
+	
+	return json.dumps(stations_details)
 
-	stations_details = {'name': station.name_human_readable, 'route_name': route_name, 'id': station.id}
+@app.route("/station/<string:station_id>/direction/<string:direction>/details", methods=['GET'])
+def get_directed_station_details(station_id, direction):
+	station = session.query(db.Station).filter(db.Station.id == station_id).one()
+	route_name = session.query(db.Route).filter(db.Route.id == station.route_id).one().name
+	
+	predictions = json.loads(get_next_service_for_station(station_id, direction))
+	
+	stations_details = {'name': station.name_human_readable, 
+						'route_name': route_name, 
+						'id': station.id,
+					   'pre_1': predictions['prediction1'],
+					   'pre_2': predictions['prediction2']}
+	
 	return json.dumps(stations_details)
 
+@app.route("/station/<string:station_id>/direction/<string:direction>/nextservice", methods=['GET'])
+def get_next_service_for_station(station_id, direction):
+	predictions = Functions.current_predictions(session, station_id)
+	
+	directed_predictions = []
+    
+	if (predictions != None):
+		for prediction in predictions:
+			trip = session.query(db.Trip).filter(db.Trip.id == prediction.trip_id).first()
+			dir = int(trip.destination_station_id > trip.origin_station_id)
+            
+			if (int(direction) is dir):
+				directed_predictions.append(prediction)
+		
+		if len(directed_predictions) is 0:
```

Comment:
Similar case as above, but reversed... recommended to use == here. Interesting reasoning that I only learned about recently! http://stackoverflow.com/a/2239753/686339